### PR TITLE
Update documentation, normalize snake_case

### PR DESCRIPTION
**Pull request by dbolack**
Document support of auto-init, change source reference to reflect being in an external module, add note about topics needing to be lower case.

Change - as a join delimiter to _ to match style guide.

### CODE REVIEW

File: managedRepository/README.md [hunk 1]
```diff
@@ -32,13 +32,14 @@ has_projects
...
 allow_squash_merge
+auto_init
 topics
 ```
 ## Example
 
 ```
 module "MyReop" {
-  source                      = "./managedRepository"
+  source                      = "github.com/enthought/terraform-modules.git//managedRepository"
```
Comment (dpinte) [inline managedRepository/README.md]: @dbolack can you also rename the directory? `managedRepository` -> `managed_repository`

File: managedRepository/main.tf [hunk 2]
```diff
@@ -17,7 +17,7 @@ resource "github_repository" "managed_repository" {
   topics             = "${var.topics}"
 }
 
-resource "github_branch_protection" "managed_repository-master_branch_protection" {
```
Comment (dpinte) [inline managedRepository/main.tf]: @dbolack we definitely need a better documentation about the conventions. One other implicit convention we developed while getting to use terraform is that resource names would use the "{}-{}" format, which is exactly what you did. Can you revert the change? 

To summarize the convention:
- we use snake_case for variable names, directory names
- compounded resource names use a `-` as delimiter

Comment (dbolack) [reply managedRepository/main.tf]: > @dbolack we definitely need a better documentation about the conventions. One other implicit convention we developed while getting to use terraform is that resource names would use the "{}-{}" format, which is exactly what you did. Can you revert the change?
> 
> To summarize the convention:
> 
> * we use snake_case for variable names, directory names
> * compounded resource names use a `-` as delimiter

I have reverted and added this to the style guide section.

File: managedRepository/README.md (hunk 1 above)
Comment (dbolack) [reply managedRepository/README.md]: Done
